tests/test_nvml_monitor.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+import time
+import sys
+sys.path.append('../model_analyzer')
+
+
+class TestNVMLMonitor(unittest.TestCase):
+
+    def test_record_memory(self):
+        from model_analyzer.monitor.nvml import NVMLMonitor
+
+        # One measurement every 0.01 seconds
+        frequency = 0.01
+        monitoring_time = 10
+        nvml_monitor = NVMLMonitor(frequency)
```

Comment:
so while this is a good test, it is not a good unit test. It requires launching the actual NVML monitor and library, which accesses external resources. This is the edge case that I talked about that is difficult to unit test. One way to handle this is to create a `TestNVMLMonitor` that derives `NVMLMonitor` but overrides all methods that actually make calls to pynvml. The overriden methods can then return known and set behavior. This way you are testing the rest of the code. However, you first need to abstract methods that make the calls to the NVML library and override them.

Reply:
I see. I will add a mock NVML class to fix this.